src/wallet/spend.cpp

```diff
@@ -767,10 +783,32 @@ static bool CreateTransactionInternal(
 
     // Get available coins
     std::vector<COutput> vAvailableCoins;
-    AvailableCoins(wallet, vAvailableCoins, &coin_control, 1, MAX_MONEY, MAX_MONEY, 0);
 
-    // Choose coins to use
-    std::optional<SelectionResult> result = SelectCoins(wallet, vAvailableCoins, /* nTargetValue */ selection_target, coin_control, coin_selection_params);
+    // First, do coin selection on a filtered subset of coins, filtered by OutputType
+    // Attempt with older output types first (LEGACY > P2SH > BECH32 > BECH32M)
+    std::optional<SelectionResult> result = [&] {
+	for (const OutputType& type : OUTPUT_TYPES) {
```

Comment:
This assumes that `OUTPUT_TYPES` is sorted by age? Maybe add a comment there to ensure this doesn't break?

Also, this is the wrong indentation? Maybe run clang-format?